src/status_im/chat/models/delete_message_for_me_test.cljs

```diff
@@ -2,114 +2,118 @@
   (:require [cljs.test :refer-macros [deftest is testing]]
             [status-im.chat.models.delete-message-for-me :as
              delete-message-for-me]
+            [status-im.chat.models.message-list :as message-list]
             [status-im.utils.datetime :as datetime]))
 
 (defonce mid "message-id")
 (defonce cid "chat-id")
 
 (deftest delete-for-me
-  (let [db      {:messages {cid {mid {:id mid}}}}
-        message {:message-id mid :chat-id cid}]
-    (testing "delete for me"
-      (let [expected  {:db {:messages {"chat-id" {"message-id"
-                                                  {:id              "message-id"
-                                                   :deleted-for-me? true}}}}
-                       :utils/dispatch-later
-                       [{:dispatch [:chat.ui/delete-message-for-me-and-sync
-                                    {:chat-id "chat-id" :message-id "message-id"}]
-                         :ms       1000}]}
-            result    (delete-message-for-me/delete {:db db} message 1000)
-            timestamp (+ (datetime/timestamp) 1000)]
-        (is (= (update-in result [:db :messages "chat-id" "message-id"] dissoc :deleted-for-me-undoable-till)
-               expected))
-        (is (-> (get-in result [:db :messages "chat-id" "message-id" :deleted-for-me-undoable-till])
-                (- timestamp)
-                js/Math.abs
-                (< 10)))))
-    (testing "should return nil if message in db"
-      (is (= (delete-message-for-me/delete {:db {:messages []}} message 1000)
-             nil)))))
+  (with-redefs [message-list/rebuild-message-list identity]
+    (let [db      {:messages {cid {mid {:id mid}}}}
+          message {:message-id mid :chat-id cid}]
+      (testing "delete for me"
+        (let [expected  {:db {:messages {"chat-id" {"message-id"
+                                                    {:id              "message-id"
+                                                     :deleted-for-me? true}}}}
+                         :utils/dispatch-later
+                         [{:dispatch [:chat.ui/delete-message-for-me-and-sync
+                                      {:chat-id "chat-id" :message-id "message-id"}]
+                           :ms       1000}]}
+              result    (delete-message-for-me/delete {:db db} message 1000)
+              timestamp (+ (datetime/timestamp) 1000)]
+          (is (= (update-in result [:db :messages "chat-id" "message-id"] dissoc :deleted-for-me-undoable-till)
+                 expected))
+          (is (-> (get-in result [:db :messages "chat-id" "message-id" :deleted-for-me-undoable-till])
+                  (- timestamp)
+                  js/Math.abs
+                  (< 10)))))
```

Comment:
If I understand this correctly, the most important part being tested here is that the key `:deleted-for-me-undoable-till` (with value timestamp + 4000) is returned. But it's being dissoc'ed in the first assert and then its value is specifically asserted in the second.

Perhaps a better way to do this would be to use `with-derefs` for `datetime/timestamp` to specify a specific timestamp for `update-db-delete-locally`, so that you can predict and test for the exact value returned.

And it might be better to use the actual `undo-time-limit-ms` value so that the test is always in sync with the code-under-test.